libtenzir/builtins/operators/api.cpp

```diff
@@ -113,6 +115,23 @@ class plugin final : public virtual operator_plugin<api_operator> {
     return std::make_unique<api_operator>(
       std::move(endpoint), request_body ? std::move(*request_body) : "{}");
   }
+
+  auto make(invocation inv, session ctx) const
+    -> failure_or<operator_ptr> override {
+    auto endpoint = located<std::string>{};
+    auto request_body = std::optional<located<record>>{};
+    argument_parser2::operator_("api")
+      .add(endpoint, "<endpoint>")
+      .add(request_body, "<request-body>")
+      .parse(inv, ctx)
+      .ignore();
+    if (not request_body) {
+      return std::make_unique<api_operator>(std::move(endpoint.inner), "{}");
+    }
+    auto request_body_json = check(to_json(request_body->inner));
```

Comment:
I know the roundtrip isn't nice, but I wanted to avoid having to change the API from accepting a JSON string for now, so this will have to do.